src/codemagic/cli/argument/argument_parser_builder.py

```diff
@@ -123,16 +133,40 @@ def _get_custom_argument_group(self, group_name) -> ArgumentGroup:
             self._custom_arguments_groups[group_name] = argument_group
         return argument_group
 
+    def _get_custom_mutually_exclusive_required_group(self, group_name: str) -> ArgumentGroup:
+        try:
+            argument_group = self._custom_arguments_groups[group_name]
+        except KeyError:
+            group_description = Colors.UNDERLINE(
+                f"One of following arguments are required for "
+                f"{Colors.BOLD(self._cli_action.action_name)} to {Colors.BOLD(group_name)}",
+            )
+            group = self._action_parser.add_argument_group(group_description)
+            argument_group = group.add_mutually_exclusive_group(required=True)
+            self._custom_arguments_groups[group_name] = argument_group
+        return argument_group
+
     def _get_argument_group(self, argument) -> ArgumentGroup:
-        if argument.argument_group_name is None:
+        if argument.argument_group_name is None and argument.mutually_exclusive_required_group is None:
```

Comment:
It would be great if something somewhere validated that argument is not assigned both to a named group and a mutually exclusive group.